tools/jit/gen_unboxing.py

```diff
@@ -0,0 +1,177 @@
+        )
+        sig = sig_group.most_faithful_signature()
+
+        if self.target is Target.DECLARATION:
+            # Note [The ATen Codegen Unboxing API]
+            # Similar to the ATen Operators API, ATen Codegen Unboxing API lives in the at::unboxing namespace, and
+            # will be used by codegen unboxing wrappers (CodegenUnboxingWrappers.cpp).
+            # The Wrappers will be registered into torch::jit::OperatorRegistry using RegisterOperators API.
+            #
+            # Important characteristics about the Codegen Unboxing API:
+            # (1) It follows the OperatorRegistry API.
+            #     This is kind of necessary to avoid overhead.
+            #     For example: if it followed the C++ API, then all of the faithful C++ factory functions
+            #     would need to wrap their arguments into TensorOptions only to unwrap them again.
+            # (2) Under the hood it calls C++ API.
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack);
+"""
+        else:
+            # gather all the arguments from native function, including "out"
+            args = list(f.func.arguments.flat_non_out) + (
+                list(f.func.arguments.out) if f.func.arguments.out else []
+            )
+
+            # parse arguments into C++ code
+            expr_list, code_list = unboxing.convert_arguments(args)
+
+            # for each C++ argument, generate the conversion code
+            code_connector = "\n\t"
+            arg_connector = ",\n\t\t"
+            # function call and push back to stack
+            prefix = "self_base." if sig.method else "at::"
+            args_str = "" if not expr_list else f"""
+        {arg_connector.join(e.expr for e in translate(expr_list, sig.arguments(), method=sig.method))}
+    """
+            ret_type: CType = cpp.returns_type(f.func.returns)
+            if isinstance(ret_type, BaseCType) and ret_type.type == voidT:
+                ret_str = ""
+                push_str = ""
+            else:
+                ret_str = "auto result_ = "
+                push_str = """
+    pack(stack, std::move(result_));
+                """
+            return f"""
+// aten::{f.func}
+TORCH_API void {f.func.name.unambiguous_name()}(Stack & stack) {{
+    {code_connector.join(code_list)}
+
+    drop(stack, {len(args)});
+
+    {ret_str}{prefix}{sig.name()}({args_str});
+    {push_str}
+}}
+"""
+
+# Generates RegisterCodegenUnboxedKernels.cpp.
+@dataclass(frozen=True)
+class ComputeCodegenUnboxedKernels:
+    @method_with_native_function
+    def __call__(self, f: NativeFunction) -> str:
+        # We unconditionally generate function wrappers,
+        sig_group = CppSignatureGroup.from_native_function(
+            f, method=(Variant.method in f.variants), fallback_binding=f.manual_cpp_binding
+        )
+
+        sig = sig_group.signature
```

Comment:
is there a reason this isn't `most_faithful_signature()`? I would've thought you'd end up with the versions of ops that takes `TensorOptions` if you used this. (you can probably sanity check by looking at the generated `RegisterCodegenUnboxedKernels.cpp`)

Reply:
Don't think it matters since we are only using the `sig.name()` but will align with the other use case.